proposals/evaluation/uc-0-effective-acr.md

```diff
@@ -0,0 +1,56 @@
+# Effective Access Control Resource Discovery Use Cases
+
+This is part 0 of the [implementation specific use cases comparison](./use-cases.md).
+
+The resource governing access over another resource is called its effective access control resource (ACR).
+
+## 1. Discovering a resource's effective access control
+
+A Solid client needs to discover effective ACR in order to understand and or edit access permissions.
+
+Both WAC and ACP use the same effective ACR discovery mechanism.
+
+### Setup
+
+We have the following hierarchy of resources:
+
+```
+</x>
+</x.acr>
+```
+
+### Universal effective ACR discovery
+
+An agent making a GET or HEAD HTTP request on `</x>` will receive a `Link: </x.acr>; rel="acl"` header in the response that points to the above `</x.acr>`.
+
+The Link header with relationship type `acl` indicates the access control resource of a resource.
+
+### ACP
+
+In ACP, every resource has exactly 1 effective access control resource directly associated with it, and every access control resource directly governs access over itself and exactly one other resource.
+
+In ACP, access control statements can be spread over several resources; that is, an access control resource can reference other resources.
+
+In ACP, the access control system in place (that is, ACP, as opposed to WAC, for example) is indicated via a Link header of `rel="type"` `<http://www.w3.org/ns/solid/acp#AccessControlResource>` in HTTP response to a request for an ACR.
+
+### WAC
+
+In WAC, the effective ACR of a resource might be the ACR of a parent container of the resource.
```

Comment:
From the client's point of view the only thing that matters should be the `Link: <....acl>; rel="acl">` relation in the header.
The text about the effective ACL process (see my comment below) is aimed at the server I think. 
But that means that the `Link: <...acl>, rel="acl"> link when pointing to the default only allows the client to edit the default header, as the client cannot just create ACL resources by guessing them.

So that means that: a server that allows the client to also edit acls for a resource must have 2 `Link:` headers: one for the default and that could be created with a PUT, in order to override the default. I think @csarven agreed with that in the last meeting.

Reply:
I don't recall saying that. I probably said that extensions can be introduced to those things - as mentioned in the current WAC ED.

Reply:
Ah I thought we had come to the conclusion that having two acl links could solve the problem. But it is not in the meeting notes. 

Does the spec explicitly state that one cannot have two link headers? 

Anway, so if I understand you correctly, the right way to do things is for the acl link header to point to the non-existent acl for that resource, and if the client gets a [404 Page not found](https://en.wikipedia.org/wiki/HTTP_404) then it should follow the links down the hierarchy container by container to find the default.

Reply:
I added a full description of what needs to be done in WAC with a default three levels deep in this commit. https://github.com/solid/authorization-panel/pull/244/commits/893dc36ff73f74e6b2eb87a03fba1910710a37c7

Reply:
Superb, thank you Henry.